diff --git a/SCV/Project.pm b/SCV/Project.pm
index 6ff2664..5f4c540 100644
--- a/SCV/Project.pm
+++ b/SCV/Project.pm
@@ -49,7 +49,7 @@ EOF
 	syswrite( $makefile_fh, $makefile );
 
 	# Use the wrapper to make sure it works
-	my $ret = system( "wrap/scv_wrap_test make -C $tmp_dir" );
+	my $ret = system( "wrap/scv_wrap_test bmake -C $tmp_dir" );
 	die "make failed: $ret\n" if ($ret);
 }
 
diff --git a/instrument/Makefile b/instrument/Makefile
index 965c8a9..ffd0cff 100644
--- a/instrument/Makefile
+++ b/instrument/Makefile
@@ -16,12 +16,13 @@ LDADD += -Wl,--start-group \
 	-lclangFrontendTool \
 	-lclangLex \
 	-lclangParse \
-	-lclangRewriteCore \
+	-lclangRewrite \
 	-lclangRewriteFrontend \
 	-lclangSema \
 	-lclangSerialization \
 	-lclangTooling \
 	-Wl,--end-group \
+	-lstdc++
 
 LDADD += `llvm-config --libs bitreader mcparser transformutils option --system-libs`
 
diff --git a/instrument/main.cxx b/instrument/main.cxx
index 28f4491..b95ef42 100644
--- a/instrument/main.cxx
+++ b/instrument/main.cxx
@@ -87,7 +87,7 @@ instrument(int argc, char *argv[], std::vector<std::string> &source_files)
 	// that will return a new MyFrontendAction object every time.  To
 	// further customize this, we could create our own factory class.
 	// int ret = Tool.run(new MFAF(inst_files));
-	int ret = Tool.run(newFrontendActionFactory<MyFrontendAction>());
+	int ret = Tool.run(&(*newFrontendActionFactory<MyFrontendAction>()));
 	if (ret)
 		errx(1, "Instrumentation failed");
 }
