diff --git a/SCV/Project.pm b/SCV/Project.pm
index 5b608cc..2d21b88 100644
--- a/SCV/Project.pm
+++ b/SCV/Project.pm
@@ -49,7 +49,7 @@ EOF
 	syswrite( $makefile_fh, $makefile );
 
 	# Use the wrapper to make sure it works
-	my $ret = system( "wrap/scv_wrap make -C $tmp_dir" );
+	my $ret = system( "wrap/scv_wrap bmake -C $tmp_dir" );
 	die "make failed: $ret\n" if ($ret);
 }
 
diff --git a/instrument/Makefile b/instrument/Makefile
index 5d3511b..9069cdf 100644
--- a/instrument/Makefile
+++ b/instrument/Makefile
@@ -18,12 +18,14 @@ LDADD += -Wl,--start-group \
 	-lclangFrontendTool \
 	-lclangLex \
 	-lclangParse \
-	-lclangRewriteCore \
+	-lclangRewrite \
 	-lclangRewriteFrontend \
 	-lclangSema \
 	-lclangSerialization \
 	-lclangTooling \
 	-Wl,--end-group \
+	-lstdc++ \
+	-lbsd
 
 LDADD += `llvm-config --libs bitreader mcparser transformutils option --system-libs`
 
diff --git a/instrument/main.cc b/instrument/main.cc
index ce10a7f..fa7d58a 100644
--- a/instrument/main.cc
+++ b/instrument/main.cc
@@ -2,6 +2,7 @@
 #include <libgen.h>
 #include <string.h>
 #include <unistd.h>
+#include <bsd/stdlib.h>			// setprogname
 
 #include <fstream>
 #include <iostream>
@@ -83,7 +84,7 @@ instrument(int argc, char *argv[], std::vector<std::string> const &source_files)
 	// that will return a new MyFrontendAction object every time.  To
 	// further customize this, we could create our own factory class.
 	// int ret = Tool.run(new MFAF(inst_files));
-	int ret = Tool.run(newFrontendActionFactory<MyFrontendAction>());
+	int ret = Tool.run(&(*newFrontendActionFactory<MyFrontendAction>()));
 	if (ret)
 		warnx("Instrumentation failed");
 	return ret;
