diff --git a/SCV/Project.pm b/SCV/Project.pm
index 106f930..bc87aea 100644
--- a/SCV/Project.pm
+++ b/SCV/Project.pm
@@ -49,7 +49,7 @@ EOF
 	syswrite( $makefile_fh, $makefile );
 
 	# Use the wrapper to make sure it works
-	my $ret = system( "wrap/scv_wrap make -C $tmp_dir" );
+	my $ret = system( "wrap/scv_wrap bsdmake -C $tmp_dir" );
 	die "make failed: $ret\n" if ($ret);
 }
 
diff --git a/instrument/Makefile b/instrument/Makefile
index 5d3511b..bea4c35 100644
--- a/instrument/Makefile
+++ b/instrument/Makefile
@@ -8,7 +8,8 @@ CXXFLAGS += `llvm-config --cxxflags`
 LDFLAGS += `llvm-config --ldflags`
 
 # Clang has a circular dependencies here, use "linker group options" to avoid
-LDADD += -Wl,--start-group \
+LDADD += \
+	-lstdc++ \
 	-lclangAST \
 	-lclangAnalysis \
 	-lclangBasic \
@@ -18,12 +19,11 @@ LDADD += -Wl,--start-group \
 	-lclangFrontendTool \
 	-lclangLex \
 	-lclangParse \
-	-lclangRewriteCore \
+	-lclangRewrite \
 	-lclangRewriteFrontend \
 	-lclangSema \
 	-lclangSerialization \
-	-lclangTooling \
-	-Wl,--end-group \
+	-lclangTooling
 
 LDADD += `llvm-config --libs bitreader mcparser transformutils option --system-libs`
 
diff --git a/instrument/instrumenter.cc b/instrument/instrumenter.cc
index aa909af..42ae0fb 100644
--- a/instrument/instrumenter.cc
+++ b/instrument/instrumenter.cc
@@ -108,14 +108,14 @@ instrumenter::real_loc_end(Stmt *d)
 
 // MyFrontendAction ---
 
-ASTConsumer *
+std::unique_ptr<ASTConsumer>
 MyFrontendAction::CreateASTConsumer(CompilerInstance &CI, StringRef file)
 {
 	// llvm::errs() << "** Creating AST consumer for: " << file << "\n";
 	SourceManager &sm = CI.getSourceManager();
 	TheRewriter.setSourceMgr(sm, CI.getLangOpts());
 
-	return new MyASTConsumer(TheRewriter);
+	return std::unique_ptr<ASTConsumer>(new MyASTConsumer(TheRewriter));
 }
 
 unsigned int
diff --git a/instrument/instrumenter.h b/instrument/instrumenter.h
index 3c2f019..08ed671 100644
--- a/instrument/instrumenter.h
+++ b/instrument/instrumenter.h
@@ -52,7 +52,7 @@ public:
 	MyFrontendAction() {};
 
 	void EndSourceFileAction() override;
-	ASTConsumer *CreateASTConsumer(CompilerInstance &CI, StringRef file);
+	std::unique_ptr<ASTConsumer> CreateASTConsumer(CompilerInstance &CI, StringRef file) override;
 
 private:
 	Rewriter TheRewriter;
diff --git a/instrument/main.cc b/instrument/main.cc
index 3b54a1b..3480838 100644
--- a/instrument/main.cc
+++ b/instrument/main.cc
@@ -80,7 +80,7 @@ instrument(int argc, char *argv[], std::vector<std::string> &source_files)
 	// that will return a new MyFrontendAction object every time.  To
 	// further customize this, we could create our own factory class.
 	// int ret = Tool.run(new MFAF(inst_files));
-	int ret = Tool.run(newFrontendActionFactory<MyFrontendAction>());
+	int ret = Tool.run(&(*newFrontendActionFactory<MyFrontendAction>()));
 	if (ret)
 		errx(1, "Instrumentation failed");
 }
diff --git a/viewer/Makefile b/viewer/Makefile
index 6de9010..9c9a1ad 100644
--- a/viewer/Makefile
+++ b/viewer/Makefile
@@ -15,9 +15,11 @@ SRCS = main.cc \
 MKDEP = `pkg-config gl glew freetype2 --cflags`
 
 CXXFLAGS += -std=c++11
-CXXFLAGS += `pkg-config gl glew freetype2 --cflags` -I/usr/local/include
+CXXFLAGS += -Wno-deprecated
+CXXFLAGS += `pkg-config glew freetype2 --cflags`
 
-LDFLAGS += `pkg-config gl glew freetype2 --libs`
-LDADD += -lstdc++ -lm -lglut glyphy/libglyphy.a
+LDFLAGS += `pkg-config glew freetype2 --libs`
+LDADD += -framework OpenGL -framework GLUT
+LDADD += -lstdc++ -lm glyphy/libglyphy.a
 
 .include <bsd.prog.mk>
